spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/mapping/SpannerObjectMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.core.mapping;
+
+import java.util.List;
+
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Struct;
+
+import org.springframework.data.convert.EntityReader;
+import org.springframework.data.convert.EntityWriter;
+
+/**
+ * Interface for mappers that can populate fields from Spanner results.
+ *
+ * @author Ray Tsang
+ * @author Chengyuan Zhao
+ */
+public interface SpannerObjectMapper
```

Comment:
Again, where's the implementation?

Reply:
I'm hoping to nail down the interfaces in this PR first. I don't think an implementation is necessary to evaluate these interfaces. This also goes along with our initial goal of just getting the template in first.

Reply:
OK, fair enough.